src/shared/dissect-image.c

```diff
@@ -2255,6 +2264,14 @@ int dissected_image_acquire_metadata(DissectedImage *m) {
         BLOCK_SIGNALS(SIGCHLD);
 
         assert(m);
+        assert(m->image_name);
+
+        if (!image_name_is_valid(m->image_name))
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Image name %s is not valid", m->image_name);
```

Comment:
i'd already filter this out when collecting the info, and suppress it if unsuitable as image name. i.e. don't even store the potentially invalid data. (and if the data is invalid, and thus image_name NULL just skip trying to read it here)